tests/eclient/testdata/dev_local_info.txt

```diff
@@ -0,0 +1,262 @@
+{{define "network"}}n1{{end}}
+{{define "ssh"}}ssh -q -o ConnectTimeout=10 -o StrictHostKeyChecking=no -o PasswordAuthentication=no -i {{EdenConfig "eden.tests"}}/eclient/image/cert/id_rsa root@{{end}}
+
+[!exec:bash] stop
+[!exec:sleep] stop
+[!exec:ssh] stop
+[!exec:chmod] stop
+[!exec:jq] stop
+
+exec chmod 600 {{EdenConfig "eden.tests"}}/eclient/image/cert/id_rsa
+
+# Starting of reboot detector with a 3 reboot limit since we reboot twice
+! test eden.reboot.test -test.v -timewait 100m -reboot=0 -count=3 &
+
+message 'Resetting of EVE'
+eden eve reset
+exec sleep 30
+
+# Create n1 network
+eden -t 1m network create 10.11.12.0/24 -n {{template "network"}}
+test eden.network.test -test.v -timewait 10m ACTIVATED {{template "network"}}
+
+# Deploy local-manager
+eden pod deploy -n local-manager --memory=512MB docker://lfedge/eden-eclient:99d7f62 -p {{template "mngr_port"}}:22 --networks={{template "network"}}
+test eden.app.test -test.v -timewait 10m RUNNING local-manager
+
+# Wait for ssh access
+exec -t 5m bash wait-ssh.sh {{template "mngr_port"}}
+
+# Start local manager application
+exec -t 1m bash local-manager-start.sh
+
+# Obtain local-manager IP address
+exec -t 2m bash get-app-ip.sh local-manager
+source .env
+
+# Configure local server
+eden controller edge-node update --device profile_server_token={{template "token"}}
+eden controller edge-node update --device local_profile_server=$app_ip:8888
+
+# STEP 1: Wait for devinfo status
+exec sleep 30
+exec -t 10m bash get-devinfo-status.sh
+stdout 'ONLINE'
+! stderr .
+
+# STEP 2: Deploy the second app
+eden pod deploy -n app1 --memory=512MB docker://lfedge/eden-eclient:9455582 -p {{template "app_port"}}:22 --networks={{template "network"}}
+test eden.app.test -test.v -timewait 10m RUNNING app1
+
+# Wait for ssh access
+exec -t 5m bash wait-ssh.sh {{template "app_port"}}
+
+# STEP 3: Wait for new devinfo status
+exec sleep 30
+exec -t 10m bash get-devinfo-status.sh
+stdout 'ONLINE'
+! stderr .
+
+# STEP 4: Request for shutdown. Check that app1 stops before local_manager
+exec -t 1m bash put-devinfo-cmd.sh COMMAND_SHUTDOWN &
+exec -t 5m bash wait-for-dev-state.sh SHUTTING_DOWN
+! stderr .
+# Could it already have HALTED? Match HALTING or HALTED
+exec -t 5m bash wait-for-app-state.sh app1 HALT
+! stderr .
+exec -t 1m bash get-appinfo-status.sh local-manager
+stdout 'RUNNING'
+! stderr .
+exec -t 5m bash wait-for-app-state.sh app1 HALTED
+! stderr .
+
+# Did local-manager halt? Can't check by asking local-manager
+exec -t 1m bash eden-pod-ps.sh local-manager
+! stdout 'RUNNING'
+! stderr .
+
+# STEP 5: test reboot via controller bringing them back up
+# send reboot command without wait
+eden controller edge-node reboot
+
+# STEP 5.1: Wait for ssh access
+exec -t 40m bash wait-ssh.sh {{template "app_port"}}
+
+# STEP 5.2: Start local manager application
+exec -t 1m bash local-manager-start.sh
+
+# STEP 5.3: Wait for new devinfo status
+exec sleep 30
```

Comment:
But we can't wait for the devinfo as reported by the local-manager since it isn't running. Do we have some function to look at the relatively new state reported by the device itself in ZDeviceState?

One thing I need to check is how EVE-OS decides to back-off the LPS API calls. It might need some patience after a device reboot since the LPS might not be running yet.

Reply:
@giggsoff I don't see any match on ZDeviceState in eden source code.

Reply:
I mean you can call script `wait-for-dev-state.sh` with ONLINE match instead of `get-devinfo-status.sh`. With this change we will use loop and will check the file inside LPS several times.

Reply:
Sure I can change that (I thought this check was before the LPS was started but it is after it is running.)